sim/battle.js

```diff
@@ -644,7 +647,10 @@ class Battle extends Dex.ModdedDex {
 		}
 
 		// @ts-ignore
-		if (effect['on' + eventid] === undefined) return relayVar;
+		let callback = effect['on' + eventid];
+		let typeofCallback = typeof callback;
+
+		if (typeofCallback === 'undefined') return relayVar;
```

Comment:
Err, `typeofCallback` doesn't seem to be reused; I'm confused about why it's a variable here.

Also is `typeof callback === 'undefined'` any different from `callback === undefined`?

Reply:
Good catch - I meant to use it below where we also check `typeof callback === 'function'`. But I think it just confuses things too much.